RFS/src/main/java/com/rfs/cms/CmsClient.java

```diff
@@ -0,0 +1,46 @@
+package com.rfs.cms;
+
+/*
+ * Client to connect to and work with the Coordinating Metadata Store.  The CMS could be implemented by any reasonable
+ * data store option (Postgres, AWS DynamoDB, Elasticsearch/Opensearch, etc).
+ */
+public interface CmsClient {
+    /*
+     * Creates a new entry in the CMS for the Snapshot's progress.  Returns true if we created the entry, and false if
+     * the entry already exists.
+     */
+    public boolean createSnapshotEntry(String snapshotName);
+
+    /*
+     * Attempt to retrieve the Snapshot entry from the CMS, if it exists; null if it doesn't
+     */
+    public CmsEntry.Snapshot getSnapshotEntry(String snapshotName);
+
+    /*
+     * Updates the status of the Snapshot entry in the CMS.  Returns true if the update was successful, and false if
+     * something else updated it before we could
+     */
+    public boolean updateSnapshotEntry(String snapshotName, CmsEntry.SnapshotStatus status);
+
+    /*
+     * Creates a new entry in the CMS for the Metadata Migration's progress.  Returns true if we created the entry, and
+     * false if the entry already exists.
+     */
+    public boolean createMetadataEntry();
+
+    /*
+     * Attempt to retrieve the Metadata Migration entry from the CMS, if it exists; null if it doesn't
+     */
+    public CmsEntry.Metadata getMetadataEntry();
+
+    /*
+     * Updates just the status field of the Metadata Migration entry in the CMS.  Returns true if the update was successful,
+     */
+    public boolean setMetadataMigrationStatus(CmsEntry.MetadataStatus status);
+
+    /*
+     * Updates all fields of the Metadata Migration entry in the CMS.  Returns true if the update was successful, and
+     * false if something else updated it before we could
+     */
+    public boolean updateMetadataEntry(CmsEntry.MetadataStatus status, String leaseExpiry, Integer numAttempts);
```

Comment:
This isn't a blocker - but trying to determine how the debugging workflow of the tool will typically operate, inspecting the worker status seems like a good high level starting point.  When the migration goes into the failure state, we will have the reason for the failure state at that time and it would be easier to inspect than log diving.  Also you could have a migration that completed, but maybe it took a while or had a large failure/retry rate.

Maybe this is better suited in another follow up task, or when we start doing more troubleshooting.

### New CMS Metadata Uber Entry:
```
{
    "currentMigration": { ... } // The current MetadataStatus
    "previousMigrations": [  // All of the previous attemptedMetadataStatus[]
        { ... }
    ]
}
```

### MetadataStatus
```
{
   "status": "PENDING|COMPLETE|FAILED|EXPIRED"
   "leaseExpiry": "{DATETIME}"
   "message": "{EXCEPTION_CALLSTACK} | {HUMAN_USEFUL_COMPLETION_DETAILS} | {EXPIRED_DETAILS}"  
}
```

Reply:
Ah, I understand now.  This makes sense and we should do it.  Will make a follow-up JIRA.